webapp/channels/src/packages/mattermost-redux/src/reducers/entities/channel_categories.ts

```diff
@@ -92,6 +95,44 @@ export function byId(state: IDMappedObjects<ChannelCategory> = {}, action: Gener
 
         return changed ? nextState : state;
     }
+    case ChannelTypes.RECEIVED_CHANNEL: {
```

Comment:
Similar to Daniel's comment below, is there a way to use a different redux action for this? Since we're just using `RECEIVED_CHANNEL`, I'd be worried that this is going to cause this part of the state to change too often as channels are loaded normally